src/libs/outsideClick/index.js

```diff
@@ -0,0 +1,33 @@
+class OutsideClick {
+    constructor() {
+        this.containerRef = null;
+        this.callback = null;
+        this.listening = false;
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    handleClick(event) {
+        if (!this.containerRef || !this.callback) return;
+        if (!this.containerRef.contains(event.target)) this.callback(this);
+    }
+
+    startListening(containerRef, callback) {
+        if (!containerRef || !callback) return;
+        this.containerRef = containerRef;
+        this.callback = callback;
+        document.addEventListener('click', this.handleClick);
+        this.listening = true;
+    }
+
+    stopListening() {
+        if (!this.listening) return;
+
+        this.listening = false;
+        document.removeEventListener('click', this.handleClick);
+        this.containerRef = null;
+        this.callback = null;
+    }
+}
+
+const outsideClick = new OutsideClick();
+export default outsideClick;
```

Comment:
I don't think this should be a singleton, it could be used for multiple components